d3a_interface/sim_results/cumulative_grid_trades.py

```diff
@@ -37,8 +52,62 @@ def update(self, area_dict, flattened_area_core_stats_dict):
             area_dict, flattened_area_core_stats_dict, self.accumulated_trades
         )
 
-    def restore_area_results_state(self, area_uuid, last_known_state_data):
+    @classmethod
+    def format_last_known_state(cls, area_uuid, last_known_state_data, area_name_uuid_map,
+                                accumulated_trades={}):
+        accumulated_trades[area_uuid] = accumulated_trades_default_dict
+        accumulated_trades[area_uuid]['children'] = []
+        accumulated_trades_map = {}
+        for state_data in last_known_state_data:
+            if state_data['areaName'] != 'External Trades':
+                for data_list in state_data['bars']:
+                    parsed_price = extract_number_from_list(data_list['priceLabel'])
+                    if data_list['targetArea'] not in accumulated_trades_map:
+                        accumulated_trades_map[data_list['targetArea']] = \
+                            accumulated_trades_default_dict
+                        accumulated_trades_map[data_list['targetArea']].update(
+                            {'name': data_list['targetArea']}
+                        )
+                    # Producer entries
+                    if "earned" in data_list['priceLabel'].split(" "):
+                        accumulated_trades_map[data_list['targetArea']].update(
+                            {'earned': parsed_price, 'produced': data_list['energy']}
+                        )
+                    # Consumer entries
+                    elif "spent" in data_list['priceLabel'].split(" "):
+                        accumulated_trades_map[data_list['targetArea']].update(
+                            {'consumedFrom': {data_list['targetArea']: data_list['energy']},
+                             'spentTo': {data_list['targetArea']: parsed_price}}
+                        )
```

Comment:
AFAIS, this could be reused in the next if block (instead of L86-96) so maybe move to a function? 
(not trying to be nitpicky here, just try to refactor for better readability)